src/openvslam/system.cc

```diff
@@ -115,11 +116,17 @@ system::system(const std::shared_ptr<config>& cfg, const std::string& vocab_file
     global_optimizer_ = new global_optimization_module(map_db_, bow_db_, bow_vocab_, cfg_->yaml_node_, camera_->setup_type_ != camera::setup_type_t::Monocular);
 
 
-    // create retifier, make optional
-    if (cfg_->yaml_node_["StereoRectifier.model"]) {
-        rectifier_ = std::make_unique<util::stereo_rectifier>(cfg_);
+    // create rectifier, make optional
+    if (cfg_->yaml_node_["StereoRectifier"]) {
+    if (cfg_->yaml_node_["StereoRectifier"]["model"]) {
+        spdlog::info("rectifier model found, going to rectify");
+        rectifier_ = std::make_unique<util::stereo_rectifier>(
+        cfg->camera_, openvslam::util::yaml_optional_ref(cfg->yaml_node_, "StereoRectifier"));
```

Comment:
Why `util::stereo_rectifier()` constructor was changed from one arguments constructor to 2-args one? If this fixes the problem that `stereo_rectifier(cfg->camera_, cfg->yaml_node_)` is not being called correctly, why don't fix it in the `stereo_rectifier()` constructor itself? BTW, it looks like it already [was made](https://github.com/stella-cv/stella_vslam/blob/main/src/stella_vslam/util/stereo_rectifier.cc#L14) in Community OpenVSLAM.

Reply:
Good call! I fixed it this way with reference to the examples like [here](https://github.com/lp-research/openvslam/blob/237dcaae25785353c3c97de3cfd835a278e86da8/example/run_euroc_slam.cc#L163) thinking it's the right way to call it but it's better to do it [stereo_rectifier side](https://github.com/lp-research/openvslam/blob/237dcaae25785353c3c97de3cfd835a278e86da8/src/openvslam/util/stereo_rectifier.cc#L12) as you suggested!